test/test_intensify/test_successive_halving.py

```diff
@@ -0,0 +1,347 @@
+                    seed=None,
+                    additional_info=None)
+        self.rh.add(config=self.config2, cost=10, time=10,
+                    status=StatusType.SUCCESS, instance_id=2,
+                    seed=None,
+                    additional_info=None)
+
+        with self.assertRaisesRegex(AssertionError, 'Cannot compare configs'):
+            intensifier._top_k(configs=[self.config2, self.config1, self.config3],
+                               k=1, run_history=self.rh)
+
+    @attr('slow')
+    def test_intensify_1(self):
+        """
+           test intensify with quality objective & cutoff as budget
+        """
+
+        def target(x, instance, seed, budget):
+            return (x['a'] + 1 + budget) / 1000.
+
+        taf = ExecuteTAFuncDict(ta=target, stats=self.stats, run_obj='quality')
+        taf.runhistory = self.rh
+
+        intensifier = SuccessiveHalving(
+            tae_runner=taf, stats=self.stats,
+            traj_logger=TrajLogger(output_dir=None, stats=self.stats),
+            rng=np.random.RandomState(12345), deterministic=True, run_obj_time=False,
+            instances=[1], initial_budget=1, max_budget=3, eta=2)
+
+        self.rh.add(config=self.config1, cost=1, time=1,
+                    status=StatusType.SUCCESS, instance_id=1,
+                    seed=0, budget=3,
+                    additional_info=None)
+
+        inc, _ = intensifier.intensify(challengers=[self.config2],
+                                       incumbent=self.config1,
+                                       run_history=self.rh,
+                                       aggregate_func=average_cost)
+
+        self.assertEqual(inc, self.config2)
+        self.assertEqual(self.stats.ta_runs, 2)
+        self.assertEqual(self.stats.inc_changed, 1)
+
+    @attr('slow')
+    def test_intensify_2(self):
+        """
+           test intensify with runtime objective and adaptive capping
+        """
+
+        def target(x):
+            time.sleep(1.5)
+            return (x['a'] + 1) / 1000.
+
+        taf = ExecuteTAFuncDict(ta=target, stats=self.stats, run_obj="runtime")
+        taf.runhistory = self.rh
+        taf.runhistory.overwrite_existing_runs = True
+
+        intensifier = SuccessiveHalving(
+            tae_runner=taf, stats=self.stats,
+            traj_logger=TrajLogger(output_dir=None, stats=self.stats),
+            rng=np.random.RandomState(12345), deterministic=True, cutoff=1,
+            instances=[1, 2], initial_budget=1, max_budget=2, eta=2, instance_order=None)
+
+        self.rh.add(config=self.config1, cost=.001, time=0.001,
+                    status=StatusType.SUCCESS, instance_id=1, seed=0,
+                    additional_info=None)
+        self.rh.add(config=self.config1, cost=.001, time=0.001,
+                    status=StatusType.SUCCESS, instance_id=2, seed=0,
+                    additional_info=None)
+
+        # config2 should be capped and config1 should still be the incumbent
```

Comment:
What about config3, should it also be capped?

Reply:
Yes. That should be too. I'll mention that.